Add support for --cache-from

exFalso opened the pull request:
https://github.com/bmuschko/gradle-docker-plugin/issues/628

cdancy commented on src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy:
```diff
@@ -156,6 +160,10 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
             buildImageCmd = buildImageCmd.withBuildArg(arg, value)
         }
 
+        if (!cacheFrom.isEmpty()) {
```
You should be able to get away with `if (cacheFrom) { ...` as groovy's truthiness check should take into account empty empty collections if I'm not mistaken.

bmuschko replied:
For the time that we still use convention mapping we should rather use the getter methods instead of the direct access to the property. For example see this: https://github.com/exFalso/gradle-docker-plugin/blob/f648a028dccd83a909571fbc983e55f3eb694e69/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy#L153-L154

cdancy commented:
What happens if the cache does not exist or is empty? Maybe add a test, or append to an existing, to confirm one way or another?

bmuschko commented:
@exFalso Are you still interested in addressing the review comments?

bmuschko commented:
I haven't seen any activity on this pull request. I am going to close this for now. Let us know if you'd like to address the review items in which case we can reopen the pull request.

exFalso commented:
@bmuschko I've been away, just arrived from holidays. Yes, I'd like to address the comments sometime in the next couple of days

daninthewoods commented:
Would be awesome to have this feature, thank you both.

exFalso commented:
@bmuschko @cdancy Addressed comments.

I added three tests, one testing the flag without anything in the cache, one testing a build without the flag but the layer in the cache (in which case the layer *isn't* used), and one that tests the flag with the layer (in which case the layer *is* used)

It was a very good idea to add tests, it turned out dockerjava has a bug with this flag! It seems to double-unmarshall the query parameter. I added an ugly workaround for this, see DockerBuildImage.groovy

The changes are pushed my master, but they don't seem to appear here, I'm assuming it's because the PR is closed. The diff is here https://github.com/bmuschko/gradle-docker-plugin/compare/master...exFalso:master

exFalso commented:
Should I resubmit this PR?

cdancy commented:
@exFalso yeah resubmit PR
